### DISCUSSION TITLE
Force redraw off image layer region

### DISCUSSION THREAD

**Discussion by manisandro** [pelicanmapping/osgearth#1868] [Support Q&A] [answered]
I'm using osgearth-3.2.0, and a custom layer derived from osgEarth::ImageLayer. The data which this custom layer renders can change over time, so I'd like to be able to invalidate a region of the globe to force redraw with the new data. I tried

    mMapNode->getTerrainEngine()->invalidateRegion({layer}, extent)

but this does not seem to trigger a redraw of the layer. Is there any way to achieve this?
Thanks!

**Comment by Shonik** [comment]
I am trying to achieve the exact same thing on the same version, invalidateRegion has no effects. Did you or someone else here found a solution ?

**Comment by gwaldron** [comment, marked as answer]
I've tested this and it seems to work in osgEarth master branch with a quick test of the wms_nexrad demo..
Do you have a cache enabled? If so an invalidation might just be loading cached data.

**Reply by Shonik** [reply]
The cache was indeed the problem thank you for your answer.

Would it be possible to keep the cache enabled so the majority of my map is loading faster but clear it for the tiles affected by the invalidateRegion function ? 
I don't know anything about the cache mechanism so maybe it's just nonsense.

**Reply by gwaldron** [reply]
Unfortunately no; the best you can do is to disable the cache on just the layer in question. Your request makes sense though, and it is a consideration for a future version of osgEarth.

**Reply by Shonik** [reply]
Disabling the cache only for the layer should work for me, thanks again for your answer.
